classlib/src/main/java/org/teavm/classlib/java/net/impl/TXHRURLConnection.java

```diff
@@ -86,7 +86,7 @@ private void performRequestIfNecessary() {
     @Async
     private native Boolean performRequest();
     private void performRequest(AsyncCallback<Boolean> callback) {
-        xhr.setOnReadyStateChange(() -> {
+        xhr.setOnReadyStateChange(e -> {
```

Comment:
So, this commit breaks backward compatibility. Can you please try to fix that? For example, leave old method and introduce the new one instead of changing signature?

Reply:
It's not backward compatibility in my opinion, but initially wrong implementation according to https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/readystatechange_event (it should be not runnable, but  generic event consumer). Still to remove native call in my code, I just need to inherit XHR from EventTarget. It's up to you to decide whether to leave incorrect implementation for backward compatilibity or change it. Adding requested changes in next commit.